### ISSUE TITLE
requirements issue

### ISSUE THREAD

**Issue by Han8931** [RockyLzy/TextDefender#4]
Hi, I am trying to install requirements but it seems many libraries are obsolete and not all requirements are listed in the txt file. 
Could you check this issue, please?

For example, I got this error
```
ERROR: Could not find a version that satisfies the requirement distro-info==0.21 (from versions: 0.10, 1.0)
ERROR: No matching distribution found for distro-info==0.21
```

**Comment by ColinLu50** [comment]
Hi, did you fix the error? I also met a lot of conflicts with installation. Could you please share your requirement.txt ?

**Comment by RockyLzy** [comment]
I have fixed the conflicts. You can download this project again




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "RockyLzy/TextDefender"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年4月11日(星期一) 晚上6:35
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;Re: [RockyLzy/TextDefender] requirements issue (Issue #4)





 
Hi, did you fix the error? I also met a lot of conflicts with installation. Could you please share your requirement.txt ?

**Comment by ColinLu50** [comment]
@RockyLzy I just download the updated project today but there are still conflicts. 
For example, 
```
The conflict is caused by:
    The user requested torch==1.8.0
    allennlp 2.4.0 depends on torch<1.9.0 and >=1.6.0
    bert-score 0.3.10 depends on torch>=1.0.0
    flair 0.9 depends on torch!=1.8 and >=1.5.0
```
I use conda to create a brand new environment with python 3.8.3

**Comment by Han8931** [comment]
The requirements provided by the authors are not 100% correct. It seems they just listed all the libraries installed on their computer so I manually installed one by one whenever I faced the issue. After finishing all these painful jobs I faced the dataset issue. I emailed them multiple times and requested about the dataset issue, but I got no answer about the dataset issue yet.

**Comment by ColinLu50** [comment]
> The requirements provided by the authors are not 100% correct. It seems they just listed all the libraries installed on their computer so I manually installed one by one whenever I faced the issue. After finishing all these painful jobs I faced the dataset issue. I emailed them multiple times and requested about the dataset issue, but I got no answer about the dataset issue yet.

Yeah, I also spent a lot of time editing the requirement file. Comment out some conflicted packages and it works. I think the author doesn't want to focus on this project.

**Comment by Han8931** [comment]
My recommendation is to use the Huggingface dataset libarary with the original TextAttack implementation with the same constraints provided in the paper (e.g., 30% perturbation ratio).

**Comment by ColinLu50** [comment]
Thanks for your advice. I will try it then.

**Comment by Han8931** [comment]
At least their model implementation seems correct. Just use their model implementation with the libraries.
